job_creator/test/test_utils.py

```diff
@@ -15,151 +15,120 @@
-        load_kubernetes_config()
+@mock.patch("jobcreator.utils.config")
+def test_config_grabbed_from_incluster(kubernetes_config):
+    load_kubernetes_config()
 
-        kubernetes_config.load_incluster_config.assert_called_once_with()
+    kubernetes_config.load_incluster_config.assert_called_once_with()
 
-    @mock.patch("jobcreator.utils.config")
-    def test_not_in_cluster_grab_kubeconfig_from_env_var(self, kubernetes_config):
-        def raise_config_exception():
-            raise ConfigException()
-
-        kubeconfig_path = mock.MagicMock()
-        kubernetes_config.load_incluster_config = mock.MagicMock(side_effect=raise_config_exception)
-        os.environ["KUBECONFIG"] = str(kubeconfig_path)
 
-        load_kubernetes_config()
-
-        kubernetes_config.load_incluster_config.assert_called_once_with()
-        kubernetes_config.load_kube_config.assert_called_once_with(config_file=str(kubeconfig_path))
-        os.environ.pop("KUBECONFIG", None)
-
-    @mock.patch("jobcreator.utils.config")
-    def test_not_in_cluster_and_not_in_env_grab_kubeconfig_from_default_location(self, kubernetes_config):
-        os.environ.pop("KUBECONFIG", None)
-
-        def raise_config_exception():
-            raise ConfigException()
+@mock.patch("jobcreator.utils.config")
+def test_not_in_cluster_grab_kubeconfig_from_env_var(kubernetes_config):
+    def raise_config_exception():
+        raise ConfigException()
 
-        kubernetes_config.load_incluster_config = mock.MagicMock(side_effect=raise_config_exception)
+    kubeconfig_path = mock.MagicMock()
+    kubernetes_config.load_incluster_config = mock.MagicMock(side_effect=raise_config_exception)
+    os.environ["KUBECONFIG"] = str(kubeconfig_path)
 
-        load_kubernetes_config()
+    load_kubernetes_config()
 
-        kubernetes_config.load_incluster_config.assert_called_once_with()
-        kubernetes_config.load_kube_config.assert_called_once_with()
+    kubernetes_config.load_incluster_config.assert_called_once_with()
+    kubernetes_config.load_kube_config.assert_called_once_with(config_file=str(kubeconfig_path))
+    os.environ.pop("KUBECONFIG", None)
 
-    def test_ensure_ceph_path_exists(self):
-        initial_path = "/tmp/ceph/mari/RBNumber/RB99999999/autoreduced/"
 
-        end_path = ensure_ceph_path_exists(initial_path)
+@mock.patch("jobcreator.utils.config")
+def test_not_in_cluster_and_not_in_env_grab_kubeconfig_from_default_location(kubernetes_config):
+    os.environ.pop("KUBECONFIG", None)
 
-        self.assertEqual(end_path, "/tmp/ceph/mari/RBNumber/unknown/autoreduced")
-        os.removedirs("/tmp/ceph/mari/RBNumber/unknown/autoreduced")
+    def raise_config_exception():
+        raise ConfigException()
 
-    def test_get_org_image_name_and_version_from_image_path_with_https(self):
-        image_path = "https://ghcr.io/fiaisis/mantid:6.9.1"
+    kubernetes_config.load_incluster_config = mock.MagicMock(side_effect=raise_config_exception)
```

Comment:
I believe you can actually just set the side_effect to the exception class and it will raise automatically, without needing the above function.

e.g. `side_effect=ConfigException`